### PR TITLE
 11704 make changes to how selection requests are sent to the labs-factfinder-api

### PR DESCRIPTION
**Pull request by horatiorosa**
### Summary
First hack at this bug:
 Selecting large amounts of census tracks or blocks causes the api to return a `ERR_FAILED 414 (Request-URI Too Long)` error. 

 - Updated the API request for geo selections from 500 to 6999 as POST requests
   - requests larger than 7000 are broken up into separate requests of 7000.
 - Moved the geoid from the request URL to the body of the request avoiding the 414 error

#### Tasks/Bug Numbers
 - Fixes [AB#11704](https://dev.azure.com/NYCPlanning/ITD/_workitems/edit/11704)

### CODE REVIEW

File: app/utils/fetch-selection-summary.js [hunk 1]
```diff
@@ -3,14 +3,34 @@ import Environment from '../config/environment';
...
-
+const SELECTION_SUMMARY_API_URL_POST = (survey = 'decennial', summaryVars = 'pop1,hunits') => `${SupportServiceHost}/summary/${survey}/${summaryVars}`;
 
 export default async function fetchSelectionSummary(survey, summaryVars, geoid) {
-  if(geoid.length) {
-    let selectionResponse = null;
-    selectionResponse = await fetch(SELECTION_SUMMARY_API_URL(survey, summaryVars, geoid));
-    selectionResponse = await selectionResponse.json();
+  let selectionResponse = {totals: { hunits: 0, pop1: 0 }};
+  // to ensure we don't receive errors from the URLs being too long, we limit the selections to 1500 geoids
```
Comment (TangoYankee) [inline app/utils/fetch-selection-summary.js:10]: should `1500` read `1000`?

**Review by TylerMatteo — CHANGES_REQUESTED** [review]

File: app/styles/modules/_m-colors.scss [hunk 2]
```diff
@@ -9,6 +9,11 @@
  $gray: #828892;
  $charcoal: #393C41;
  $dark-gray: #5D626A;
+ $red: #D13823;
+ $orange: #D96B27;
+ $light-orange: #FBE4D7;
+ $teal: #2C7A7B;
+;
```
[the remark this answers is not in the document]
Comment (horatiorosa) [reply app/styles/modules/_m-colors.scss:16]: oops! will rename and remove xtra `;`

File: app/templates/components/map-utility-box.hbs [hunk 3]
```diff
@@ -17,12 +28,20 @@
...
-        selected
+        {{#if (lt this.selectionCount 500 )}}
+          selected
+        {{/if}}
       </div>
       <div class="map-utility-box-data">
         <div>
           <span>
-            {{fa-icon icon='user-friends'}}
+            <svg width="60" height="60" viewBox="0 0 60 60" fill="none" xmlns="http://www.w3.org/2000/svg">
```
[the remark this answers is not in the document]
Comment (horatiorosa) [reply app/templates/components/map-utility-box.hbs:38]: No, these are icons from Chakra that Design requested to be used.

File: app/utils/fetch-selection-summary.js (hunk 1 above)
Comment (horatiorosa) [reply app/utils/fetch-selection-summary.js]: 500, hehe

File: app/templates/map-to-explorer.hbs [hunk 4]
```diff
@@ -0,0 +1,10 @@
+<div class="spinner">
+  <p>new loading</p>
+  {{log 'are we transitioning From Map Page? 'this.transitioningFromMapPage}}
+  {{#if this.transitioningFromMapPage }}
+    <p>we are transitioning from the homepage</p>
+  {{/if}}
```
[the remark this answers is not in the document]
Comment (horatiorosa) [reply app/templates/map-to-explorer.hbs]: gone gone gone

File: app/utils/fetch-selection-summary.js (hunk 1 above)
Comment (TangoYankee) [reply app/utils/fetch-selection-summary.js]: I can't keep up!

**Review by TangoYankee — CHANGES_REQUESTED** [review]
I'm able to get the user feedback message when breaking an existing selection into smaller geography. However, I can't get the message to appear when starting a fresh selection with the polygon or radius selectors.

File: app/utils/fetch-selection-summary.js [hunk 5]
```diff
@@ -2,15 +2,30 @@ import Environment from '../config/environment';
...
 export default async function fetchSelectionSummary(survey, summaryVars, geoid) {
-  if(geoid.length) {
-    let selectionResponse = null;
-    selectionResponse = await fetch(SELECTION_SUMMARY_API_URL(survey, summaryVars, geoid));
-    selectionResponse = await selectionResponse.json();
+  let selectionResponse = {totals: { hunits: 0, pop1: 0 }};
+  // to ensure we don't receive errors from the URLs being too long, we limit the selections to 500 geoids
+  // The limit can be increased when indices are added to the geoids in the pff db
+  if (geoid.length > 500) {
+    selectionResponse = null;
```
[the remark this answers is not in the document]
Comment (horatiorosa) [reply app/utils/fetch-selection-summary.js]: @TylerMatteo 
Using the POLYGON and RADIUS  and exceeding the max getoid selection limit will not generate the map-utility-box template (displaying the selection count, error message and the clear selection link) without returning `selectionResponse = null` here.

This explains why I never encountered the issue that Tim discovered until after this change. 100 internets to  @TangoYankee  for finding this bug in the big fix.

**Pull request closed without merging**